components/exercises/chapter-4-oscar/Utils.tsx

```diff
@@ -70,3 +72,139 @@ export function reverseArrayInPlace(arr: any[]): undefined {
     // After the first iteration, the array becomes [5, 2, 3, 4, 1]
   }
 }
+
+/**
+ * Represents the properties of a list item.
+ */
+export type ListItemProp = {
+  value: ReactNode;
+  rest: ListItemProp | null;
+};
+
+/**
+ * Converts an array of numbers into a linked list.
+ * @param arr - The array of numbers to be converted.
+ * @returns The resulting linked list.
+ */
+export function arrayToList(arr: ReactNode[]): ListItemProp | null {
```

Comment:
`arr` does not need to be a react node, it can be an array of strings or numbers

Reply:
refactored, thank you